### DISCUSSION TITLE
Naming generated types to avoid duplicate names in model/domain layer.

### DISCUSSION THREAD

**Discussion by marceloverdijk** [Netflix/dgs-framework#20] [Q&A] [answered]
First of all, awesome to opensource the DGS framework 👏 
This was a real missing piece bridging low-level `graphql-java` and `spring-boot`.

As I read the framework is used extensively at Netflix and I assume the codegen as well.
The examples just have the code generated model, but no real persistent storage behing him, and I wonder what pattern is used if that would be the case.

What I have done in other GraphQL (or REST) projects is to suffix these GraphQL or REST data objects with e.g. `Type` or `Resource` (e.g. `ShowType` with GraphQL or `ShowResource` with REST).

This to avoid duplicate names with the application's model/domain class `Show` (like a JPA entity class). And then use e.g. MapStruct to convert from the `Show` model class to `ShowType`.

As the DGS codegen does not allow suffixing the generated type names, I wonder how this is implemented in a more real application? Just use the same name in model layer and GraphQL layer, and use fully qualified class names? It is trivial to do but I wonder what the experience is at Netflix regarding this.

**Comment by paulbakker** [comment, marked as answer]
We have a broad variety of use cases; some teams getting data from gRPC services, some loading from a relational database, some other data store, or using some sort of mapper, etc.. 

Because of this variety we haven't seen an issue with the naming in a consistent way. I think when names overlap the FQN is used. 

That doesn't mean we can't add a feature to solve this better. I think it makes sense to add a prefix/postfix setting to codegen. It does seem like a fairly common issue.

**Reply by marceloverdijk** [reply]
Thx @paulbakker I will add a feature request.

**Reply by marceloverdijk** [reply]
https://github.com/Netflix/dgs-codegen/issues/21

**Reply by Ramblurr** [reply]
The issue link above is stale.. here is the current one https://github.com/Netflix/dgs-codegen/issues/323
